ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/SubmitterInfo.scala

```diff
@@ -26,12 +27,37 @@ import java.time.Instant
   *   command deduplication. If it chooses to do so, it must follow the
   *   same rules as the participant:
   *   - Deduplication is based on the (submitter, commandId) tuple.
-  *   - Commands must not be deduplicated after the `deduplicateUntil` time has passed.
+  *   - Commands must not be deduplicated after the deduplicateUntil time has passed.
   *   - Commands should not be deduplicated after the command was rejected.
   */
 final case class SubmitterInfo(
+    actAs: List[Party],
+    applicationId: ApplicationId,
+    commandId: CommandId,
+    deduplicateUntil: Instant,
+) {
+  // Note: this function is only available temporarily until the entire DAML code base
+  // supports multi-party submissions. Use at your own risk.
+  def singleSubmitterOrThrow(): Party = {
+    if (actAs.length == 1)
+      actAs.head
+    else
+      throw new RuntimeException("SubmitterInfo contains more than one acting party")
```

Comment:
Can we have a real exception type for this?

```
final class UnexpectedMultipleSubmittersException
  extends RuntimeException("SubmitterInfo contains more than one acting party")
```

Reply:
Out of interest, what is the benefit of having a real exception?

For reference, this entire block should be deleted before we enable multi-party submissions on the ledger API.